google/generativeai/caching.py

```diff
@@ -111,17 +145,18 @@ def _prepare_create_request(
         if contents:
             contents = content_types.to_contents(contents)
```

Comment:
Implicit role assignment also needed here
```suggestion
            contents = content_types.to_contents(contents)
            if not contents[-1].role:
                contents[-1].role = _USER_ROLE   # define _USER_ROLE at the top
```